api/v2/views/base.py

```diff
@@ -157,8 +164,24 @@ def perform_create(self, serializer):
     @unresolved_requests_only
     def destroy(self, request, *args, **kwargs):
         instance = self.get_object()
-        self.perform_destroy(instance)
-        return Response(status=status.HTTP_204_NO_CONTENT)
+        try:
+            self.perform_destroy(instance)
+            return Response(status=status.HTTP_204_NO_CONTENT)
+        except Exception as e:
+            message = {
+                "An error was encoutered when closing the request: %s" % e.message
+            }
+            logger.exception(e)
+            raise exceptions.ParseError(detail=message)
+
+    def perform_destroy(self, instance):
+        """
+        Add an end date to a request and take no further action
+        """
+        status, _ = StatusType.objects.get_or_create(name="closed")
+        instance.status = status
```

Comment:
This is okay since, `Machine/Resource` are subtypes, and so far we havn't needed statuses that are unique to each.